addons/mail/tools/discuss.py

```diff
@@ -200,7 +201,19 @@ def many_ids(records, mode="REPLACE", /, *, as_thread=False):
         should be avoided. It is kept as a public method until all remaining
         occurences can be removed.
         Using the method ``many(..., only_id=True)`` is preferable."""
-        res = [Store.one_id(record, as_thread=as_thread) for record in records]
+        if records._name == "mail.message.reaction":
+            res = []
+            for content, reactions_list in groupby(
+                    records, lambda r: r.content
+            ):
```

Comment:
I think you're missing grouping by `message_id` here, it works by luck because it's only called with reactions of a single message.

Reply:
It was intended because I thought logically we don't call reactions of several messages. I can add it there but I don't know what is its probably use case.